account_credit_control/models/res_partner.py

```diff
@@ -67,3 +66,15 @@ def _check_credit_policy(self):
             except UserError as err:
                 # constrains should raise ValidationError exceptions
                 raise ValidationError(err)
+
+    def write(self, vals):
+        # There is a bug with Odoo ...
+        # The field "credit_policy_id" is considered as an "old field" and
+        # the field property_account_receivable_id like a "new field"
+        # the default check with constrain are not possible as fields are
+        # updated not in the same time on ORM write
+        # so we call check when all fields written to the record
+        res = super(ResPartner, self).write(vals)
+        if vals.get("credit_policy_id") and vals.get("property_account_receivable_id"):
```

Comment:
```suggestion
        if vals.get("credit_policy_id") or vals.get("property_account_receivable_id"):
```
As I understand, we need an `or` here.

Reply:
no, once write method is triggered with the new policy and old account so it generates an error
actually, we have a domain on policy and onchange with parents user from wrong input anyway

Reply:
But there is then a functional issue that you're introducing.
AFAIU if one writes a `credit_policy_id` first and writes `property_account_receivable_id` after, the check will never be done, and we won't respect the constraint...

Reply:
UPD updated to "or" operator